kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/KafkaProxyIngressReconcilerTest.java

```diff
@@ -101,21 +119,100 @@ void shouldSetResolvedRefsToTrueWhenProxyFound() throws Exception {
 
     }
 
+    @SuppressWarnings("OptionalGetWithoutIsPresent")
     @Test
-    void shouldSetResolvedRefsToUnknown() {
+    void shouldAddChecksumForValidProxyRef() throws Exception {
         // given
         Clock z = Clock.fixed(Instant.EPOCH, ZoneId.of("Z"));
         var reconciler = new KafkaProxyIngressReconciler(z);
 
-        Context<KafkaProxyIngress> context = mock(Context.class);
+        when(context.getSecondaryResource(KafkaProxy.class, KafkaProxyIngressReconciler.PROXY_EVENT_SOURCE_NAME)).thenReturn(Optional.of(PROXY));
+
+        // when
+        var update = reconciler.reconcile(INGRESS, context);
+
+        // then
+        assertThat(update)
+                .isNotNull()
+                .satisfies(uc -> Assertions.assertThat(update.getResource())
+                        .isPresent()
+                        .satisfies(kpi -> OperatorAssertions.assertThat(kpi.get())
+                                .hasAnnotationSatisfying(ResourcesUtil.DEPENDANT_CHECKSUM_ANNOTATION,
+                                        actualValue -> Assertions.assertThat(actualValue)
+                                                .isNotBlank()
+                                                .satisfies(str -> Assertions.assertThat(Long.parseLong(str)).isNotZero()))));
+
+    }
+
+    @SuppressWarnings("OptionalGetWithoutIsPresent")
+    @Test
+    void shouldAddExpectedChecksum() throws Exception {
+        // given
+        Clock z = Clock.fixed(Instant.EPOCH, ZoneId.of("Z"));
```

Comment:
Let's make this a constant

Reply:
Yeah, I did that in one place/test this one must have slipped past.